File: Server/route/ctfTeam.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCTFTeamRoute(r *gin.Engine) {
	adminCTFTeam := r.Group("/ctfTeam")
	adminCTFTeam.Use(middleware.AdminAuthMiddleware())
	{
		adminCTFTeam.POST("/GetCTFTeamList", api.GetCTFTeamList)
		adminCTFTeam.POST("/GetCTFTeam", api.GetCTFTeam)
		adminCTFTeam.POST("/CreateCTFTeam", api.CreateCTFTeam)
		adminCTFTeam.POST("/UpdateCTFTeam", api.UpdateCTFTeam)
		adminCTFTeam.POST("/DeleteCTFTeam", api.DeleteCTFTeam)
	}
}

File: Web/src/api/ctfTeam.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFTeamList(data) {
  return apiClient.post("/ctfTeam/GetCTFTeamList", data).then(handleResponse).catch(handleError)
}

function GetCTFTeam(data) {
  return apiClient.post("/ctfTeam/GetCTFTeam", data).then(handleResponse).catch(handleError)
}

function CreateCTFTeam(data) {
  return apiClient.post("/ctfTeam/CreateCTFTeam", data).then(handleResponse).catch(handleError)
}

function UpdateCTFTeam(data) {
  return apiClient.post("/ctfTeam/UpdateCTFTeam", data).then(handleResponse).catch(handleError)
}

function DeleteCTFTeam(data) {
  return apiClient.post("/ctfTeam/DeleteCTFTeam", data).then(handleResponse).catch(handleError)
}

const ctfTeamApi = {
  GetCTFTeamList,
  GetCTFTeam,
  CreateCTFTeam,
  UpdateCTFTeam,
  DeleteCTFTeam
}

export default ctfTeamApi
